airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java

```diff
@@ -146,12 +147,14 @@ public boolean existingSchemaMatchesStreamConfig(final StreamConfig stream, fina
         .collect(LinkedHashMap::new,
             (map, column) -> map.put(column.getKey(), column.getValue()),
             LinkedHashMap::putAll);
-    final boolean sameColumns = actualColumns.equals(intendedColumns)
+    final boolean countPksWithoutIndex = existingTable.columns().entrySet().stream()
+        .noneMatch(c -> pks.contains(c.getKey()) && !c.getValue().contains("NOT NULL"));
```

Comment:
This doesn't work for snowflake - we don't currently have 'nullness' in `SnowflakeTableDefinition` ([link](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeTableDefinition.java#L15))